scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1231,8 +1232,19 @@ def eigs(A, k=6, M=None, sigma=None, which='LM', v0=None,
             import warnings
             warnings.warn('M does not have the same type precision as A. '
                           'This may adversely affect ARPACK convergence')
+
     n = A.shape[0]
 
+    if k >= n - 1:
+        import warnings
+        warnings.warn('k greater than/equal to N - 1 for N * N square matrix. '
+                      'Using scipy.linalg.eig instead.',
+                      SparseEfficiencyWarning)
+        if return_eigenvectors:
+            return eig(A.todense())
+        else:
+            return eig(A.todense(), right=False)
```

Comment:
These can probably be condensed into one return:
```python
return eig(A.todense(), right=return_eigenvectors)
```
(Same for the similar lines below.)